SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -222,13 +257,27 @@ void FSpatialNetGUIDCache::RemoveEntityNetGUID(Worker_EntityId EntityId)
 	UClass* Class = Actor->GetClass();
 	FClassInfo& Info = SpatialNetDriver->TypebindingManager->FindClassInfoByClass(Class);
 
+	improbable::UnrealMetadata* UnrealMetadata = SpatialNetDriver->StaticComponentView->GetComponentData<improbable::UnrealMetadata>(EntityId);
+
+	if (UnrealMetadata == nullptr)
+	{
+		return;
+	}
+
+	const FUnrealObjectRef& StablyNamedRef = UnrealMetadata->StablyNamedRef;
+
 	for (auto& SubobjectInfoPair : Info.SubobjectInfo)
 	{
 		FUnrealObjectRef SubobjectRef(EntityId, SubobjectInfoPair.Key);
 		if (FNetworkGUID* SubobjectNetGUID = UnrealObjectRefToNetGUID.Find(SubobjectRef))
 		{
 			NetGUIDToUnrealObjectRef.Remove(*SubobjectNetGUID);
 			UnrealObjectRefToNetGUID.Remove(SubobjectRef);
+
+			if (StablyNamedRef != SpatialConstants::NULL_OBJECT_REF)
+			{
+				UnrealObjectRefToNetGUID.Remove(FUnrealObjectRef(0, 0, SubobjectInfoPair.Value->SubobjectName.ToString(), StablyNamedRef));
+			}
 		}
 	}
 
```

Comment:
Remove actor's stably named ref from `UnrealObjectRefToNetGUID` if it's not null.